### ISSUE TITLE
Automatic disabled apps are not updated from the app store

### ISSUE THREAD

**Issue by nickvergessen** [nextcloud/server#5933]
### Steps to reproduce
1. Upgrade from 11 to 12: `occ upgrade --no-app-disable --verbose`

Output:
```
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
2017-07-31T10:26:42+02:00 Set log level to debug
2017-07-31T10:26:42+02:00 Turned on maintenance mode
2017-07-31T10:26:42+02:00 Repair step: Repair MySQL collation
...
2017-07-31T10:26:45+02:00 Updating database schema
2017-07-31T10:26:48+02:00 Updated database
2017-07-31T10:26:48+02:00 Disabled incompatible app: files_accesscontrol
2017-07-31T10:26:48+02:00 Disabled incompatible app: files_automatedtagging
2017-07-31T10:26:48+02:00 Updating <federatedfilesharing> ...
2017-07-31T10:26:48+02:00 Updated <federatedfilesharing> to 1.2.0
...
2017-07-31T10:26:50+02:00 Updating <files_sharing> ...
2017-07-31T10:26:50+02:00 Repair step: Fix the share type of guest shares when migrating from ownCloud
2017-07-31T10:26:50+02:00 Repair step: Copy the share password into the dedicated column
2017-07-31T10:26:51+02:00 Updated <files_sharing> to 1.4.0
2017-07-31T10:26:51+02:00 Updating <files_trashbin> ...
2017-07-31T10:26:51+02:00 Updated <files_trashbin> to 1.2.0
2017-07-31T10:26:51+02:00 Updating <files_versions> ...
2017-07-31T10:26:51+02:00 Updated <files_versions> to 1.5.0
...
2017-07-31T10:26:53+02:00 Checking for update of app files in appstore
2017-07-31T10:26:53+02:00 Checked for update of app "files" in appstore 
2017-07-31T10:26:53+02:00 Checking for update of app files_markdown in appstore
2017-07-31T10:26:53+02:00 Checked for update of app "files_markdown" in appstore 
2017-07-31T10:26:53+02:00 Checking for update of app files_pdfviewer in appstore
2017-07-31T10:26:53+02:00 Checked for update of app "files_pdfviewer" in appstore 
2017-07-31T10:26:53+02:00 Checking for update of app files_sharing in appstore
2017-07-31T10:26:53+02:00 Checked for update of app "files_sharing" in appstore 
2017-07-31T10:26:53+02:00 Checking for update of app files_texteditor in appstore
2017-07-31T10:26:53+02:00 Checked for update of app "files_texteditor" in appstore 
2017-07-31T10:26:53+02:00 Checking for update of app files_trashbin in appstore
2017-07-31T10:26:53+02:00 Checked for update of app "files_trashbin" in appstore 
2017-07-31T10:26:53+02:00 Checking for update of app files_versions in appstore
2017-07-31T10:26:53+02:00 Checked for update of app "files_versions" in appstore 
2017-07-31T10:26:53+02:00 Checking for update of app files_videoplayer in appstore
2017-07-31T10:26:53+02:00 Checked for update of app "files_videoplayer" in appstore 
...
2017-07-31T10:26:54+02:00 Repair step: Repair MySQL collation
....
2017-07-31T10:27:10+02:00 Starting code integrity check...
2017-07-31T10:27:14+02:00 Finished code integrity check
2017-07-31T10:27:14+02:00 Update successful
2017-07-31T10:27:14+02:00 Turned off maintenance mode
2017-07-31T10:27:14+02:00 Reset log level
```

The apps `files_accesscontrol` and `files_automatedtagging` contain storage wrappers and are therefor called early(?) and then disabled because they are incompatible. This is okay, but they should be automatically updated later and re-enabled, especially because I picked `--no-app-disable`.

The check for appstore updates however doesn't take them into account because they are disabled at that point in time.

**Comment by enoch85** [comment]
Shouldn't this be backported as well?

**Comment by nickvergessen** [comment]
We have to see how big the patch is, and if its okay, the PR should be backported, but not the issue 😆

**Comment by MorrisJobke** [comment]
Fixed with #10508

**Comment by MorrisJobke** [comment]
@nickvergessen Right?
